src/cli/commands/force/lightning/lwc/test/ui/mobile/run.ts

```diff
@@ -0,0 +1,185 @@
+                    Run.createError('error:specPathInvalidDescription')
+                );
+            }
+            let specsArray = this.enumerateTestSpecs(specPath);
+            specsArray = specsArray.map((spec) => `'${spec}'`);
+            specs = specsArray.join(' ');
+        }
+
+        CommonUtils.startCliAction(messages.getMessage('runningUtamTest'));
+        return await this.executeRunUtamTest(configPath, specs)
+            .then((result) => {
+                CommonUtils.stopCliAction();
+
+                this.logger.info(
+                    `UTAM test ran successfully:\n${result.stdout}`
+                );
+                console.log(result.stdout);
+
+                return Promise.resolve();
+            })
+            .catch((error) => {
+                CommonUtils.stopCliAction();
+
+                this.logger.warn(`Failed to run UTAM test: ${error}`);
+                console.log(error);
+
+                return Promise.reject(
+                    new SfError(
+                        util.format(
+                            messages.getMessage(
+                                'error:unexpectedErrorDescription'
+                            ),
+                            error
+                        ),
+                        'lwc-dev-mobile',
+                        Run.examples
+                    )
+                );
+            });
+    }
+
+    private async executeRunUtamTest(
+        configPath: string,
+        specs: string
+    ): Promise<any> {
+        let cmd = `npx --no-install wdio '${configPath}'`;
+        if (specs.length > 0) {
+            // If the spec flag is set then honor that and use that test over
+            // specs that are specified in the config.
+            cmd = cmd + ` --spec ${specs}`;
+        }
+
+        return CommonUtils.executeCommandAsync(cmd);
+    }
+
+    /**
+     * Recursively collects paths of test files given a starting point.
+     * If a valid test file is selected as a starting point then it will
+     * be the sole path returned in an array. Otherwise, the array returned
+     * will be collection of files' paths including subfolders' files.
+     *
+     * @param path Path to a test file or a folder that contains tests.
+     * @returns Array of paths to test files.
+     */
+    private enumerateTestSpecs(path: string): Array<string> {
+        let files: Array<string> = [];
+        const stat = fs.statSync(path);
+        if (stat.isFile()) {
+            files.push(`${path}`);
+        } else {
+            const items = fs.readdirSync(path, {
+                withFileTypes: true
+            });
+            items.forEach(async (item) => {
+                if (item.isDirectory()) {
+                    files = [
+                        ...files,
+                        ...this.enumerateTestSpecs(`${path}/${item.name}`)
+                    ];
+                } else {
+                    if (item.name.endsWith('.spec.js')) {
```

Comment:
Should the test only be `*.spec.js` or will we allow any Javascript files?

Reply:
I think we should include `*.spec.js` and `*.test.js` Or better yet, test it out with WDIO and see what happens if the spec in your config file points to a folder with files of different types/names. Would it include all files, or just `js` files (and how does it behave on the file names)?

Reply:
Btw you should probably do `item.name.toLowerCase().endsWith('.spec.js')`

Reply:
`js` works. It could be any Javascript. And we won't be able to control over that. I imagine you would say let WDIO just handle it. I am fine with that if that's the case. And I'll remove the filter.

Reply:
Yeah, I'd like us to take the simplest approach if possible. However we may not be able to just simply pass in all the files in a folder to WDIO (what if there are files in there that are not tests/specs, for example `.DS_Store`). That's why I said lets test it out.

We can create a config file that points to a folder with a bunch of files that are .js and not .js and see what does WDIO do and we can do the same?

Reply:
Experimenting with WDIO shows that WDIO will accept any `js` file as a test script. On the other hand, if a file does not end with `js` then with its own pattern matching it identifies that file is not compatible and skips the test. So this is becoming a recurring question. Do we do anything in this `run` command of just dump everything to WDIO and have it take care input to itself?

Reply:
On one hand it would be nice to simplify the code and just dump everything on WDIO to take care of. On the other hand, I'm worried that perhaps there is a limit on the total length of the commandline command (specially on windows). So if a folder has a lot of files but only a few are actual `js` files, and if we don't filter and just include everything then we could end up with a very lengthy command. But then again maybe we can safely assume that a user should not have such a folder? If we can assume that then I'd say let WDIO take care of things.

@khawkins what are your thoughts?

Reply:
I support keeping it simple, assuming rational intentions from our users, and adjust if and when those assumptions break down.